backend/Alpaki.Logic/Features/Invitations/RegisterVolunteer/RegisterVolunteerHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Alpaki.CrossCutting.Enums;
+using Alpaki.Database;
+using Alpaki.Database.Models;
+using Alpaki.Logic.Services;
+using MediatR;
+using Microsoft.AspNet.Identity;
+
+namespace Alpaki.Logic.Features.Invitations.RegisterVolunteer
+{
+    public class RegisterVolunteerHandler : IRequestHandler<RegisterVolunteer, RegisterVolunteerResponse>
+    {
+        private readonly IDatabaseContext _dBContext;
+        private readonly JwtGenerator _jwtGenerator;
+        private readonly IPasswordHasher _passwordHasher;
+
+        public RegisterVolunteerHandler(IDatabaseContext dBContext, JwtGenerator jwtGenerator)
+        {
+            _dBContext = dBContext;
+            _jwtGenerator = jwtGenerator;
+            _passwordHasher = new PasswordHasher();
+        }
+        public async Task<RegisterVolunteerResponse> Handle(RegisterVolunteer request, CancellationToken cancellationToken)
+        {
+            var invitation = _dBContext.Invitations.SingleOrDefault(
+                x => x.Email.ToLower().Equals(request.Email.ToLower())
+                     && x.Status == InvitationStateEnum.Pending
+            );
+            
+            if(invitation is null)
+                throw new Exception("Not valid invitation was found");
+
+            if ((invitation.Timestamp - DateTimeOffset.UtcNow) > TimeSpan.FromHours(24))
+                throw new Exception("Invitation has expired");
+
+            if (invitation.Attempts >= 3)
+            {
+                throw new Exception("Code is no longer valid");
+            }
```

Comment:
przerzuć to wszystko do fluentValidatora

Reply:
> przerzuć to wszystko do fluentValidatora

Powinno to na pewno być tam? Bo jakby nie patrzeć to jest logika biznesowa, a nie samo sprawdzenie czy parametry, które przyszły są poprawne

Reply:
w sumie wszystko jest logiką biznesową :) jeśli przerzucisz to do validatora będziemy mieli elegancki error response. w tej chwili poleci błąd 500 co sugeruje NASZ BŁAD a przecież zły kod to 400 ich błąd :P